tests/integration/telemetry/stackdriver/stackdriver_filter_test.go

```diff
@@ -96,28 +98,46 @@ func unmarshalFromTemplateFile(file string, out proto.Message, clName string) er
 func TestStackdriverMonitoring(t *testing.T) {
 	framework.NewTest(t).
 		Run(func(ctx framework.TestContext) {
-			sendTraffic(t)
-
-			for _, cl := range ctx.Clusters() {
-				scopes.Framework.Infof("Validating Telemetry for Cluster %v", cl)
-				retry.UntilSuccessOrFail(t, func() error {
-					clName := cl.Name()
-					//Validate cluster names in telemetry below once https://github.com/istio/istio/issues/28125 is fixed.
-					if err := validateMetrics(t, serverRequestCount, clientRequestCount, clName); err != nil {
-						return err
-					}
-					if err := validateLogs(t, serverLogEntry, clName); err != nil {
-						return err
-					}
-					if err := validateTraces(t); err != nil {
-						return err
-					}
-					if err := validateEdges(t, clName); err != nil {
+			g, _ := errgroup.WithContext(context.Background())
+			for _, cltInstance := range clt {
+				scopes.Framework.Infof("Validating Telemetry for Cluster %v", cltInstance)
+				g.Go(func() error {
+					err := retry.UntilSuccess(func() error {
+						if err := sendTraffic(t, cltInstance); err != nil {
+							return err
+						}
+						clName := cltInstance.Config().Cluster.Name()
+						//Validate cluster names in telemetry below once https://github.com/istio/istio/issues/28125 is fixed.
+						if err := validateMetrics(t, serverRequestCount, clientRequestCount, clName); err != nil {
+							return err
+						}
+						t.Logf("Metrics validated")
+						if err := validateLogs(t, serverLogEntry, clName); err != nil {
+							return err
+						}
+						t.Logf("logs validated")
+						if err := validateTraces(t); err != nil {
+							return err
+						}
+						t.Logf("Traces validated")
+						if err := validateEdges(t, clName); err != nil {
+							return err
+						}
+						t.Logf("Edges validated")
+
+						return nil
+
+					}, retry.Delay(3*time.Second), retry.Timeout(80*time.Second))
```

Comment:
can you change this back to `retry.Delay(10*time.Second), retry.Timeout(2*time.Minute))` and see if it fixes your issue

Reply:
Sure! this might be the reason it pass with GKE but failed when using Kind!